### PR TITLE
DTSPB-1576: Lookup lev record by system number

### PR DESCRIPTION
**Pull request by nigeldunne**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSPB-1576

### Change description ###
Lookup lev record by system number

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/probate/controller/LifeEventController.java [hunk 1]
```diff
@@ -30,4 +34,23 @@
...
+    @PostMapping(path = "/updateWithSystemNumber")
+    public ResponseEntity<CallbackResponse> updateWithSystemNumber(@RequestBody CallbackRequest request) {
+        final Integer systemNumber = request.getCaseDetails().getData().getDeathRecordSystemNumber();
+        final DeathRecord deathRecord = lifeEventService.getDeathRecordById(systemNumber);
+        final CallbackResponse response = callbackResponseTransformer.updateTaskList(request);
+        response.getData().setDeathRecord(deathRecord);
+        return ResponseEntity.ok(response);
+    }
+
+    @PostMapping(path = "/updateWithSystemNumber2")
```
Comment (isha2040) [inline src/main/java/uk/gov/hmcts/probate/controller/LifeEventController.java]: can you give some more relevant name to this mapping . Also a brief comment on what it does would help may be

File: src/main/java/uk/gov/hmcts/probate/model/ccd/raw/request/CaseData.java [hunk 2]
```diff
@@ -532,6 +532,8 @@
     private String registryAddress;
     private String registryEmailAddress;
     private final List<CollectionMember<DeathRecord>> deathRecords;
+    private Integer deathRecordSystemNumber;
```
Comment (isha2040) [inline src/main/java/uk/gov/hmcts/probate/model/ccd/raw/request/CaseData.java]: should these variables be final ?

File: src/main/java/uk/gov/hmcts/probate/model/ccd/raw/response/ResponseCaseData.java [hunk 3]
```diff
@@ -323,6 +323,7 @@
     private List<CollectionMember<ParagraphDetail>> paragraphDetails = new ArrayList<>();
     private String bulkScanCaseReference;
     private String grantDelayedNotificationIdentified;
-    private final List<CollectionMember<DeathRecord>> deathRecords;
-
+    private List<CollectionMember<DeathRecord>> deathRecords;
+    private Integer deathRecordSystemNumber;
```
Comment (isha2040) [inline src/main/java/uk/gov/hmcts/probate/model/ccd/raw/response/ResponseCaseData.java]: same , shouldnt these both new vars be final ?

File: src/main/java/uk/gov/hmcts/probate/service/DeathRecordService.java [hunk 4]
```diff
@@ -42,6 +42,26 @@ private DeathRecord mapDeathRecord(V1Death v1Death) {
...
+                .dateOfBirth(deceased.getDateOfBirth())
+                .sex(null == deceased.getSex() ? null : deceased.getSex().getValue())
+                .address(deceased.getAddress())
+                .dateOfDeath(deceased.getDateOfDeath());
+        }
+
+        return builder.build();
+    }
+
+    @SuppressWarnings("squid:S2583")
```
Comment (isha2040) [inline src/main/java/uk/gov/hmcts/probate/service/DeathRecordService.java:56]: is it possible to resolve this warning than suppress

File: src/main/java/uk/gov/hmcts/probate/service/LifeEventService.java [hunk 5]
```diff
@@ -41,6 +45,24 @@ public LifeEventService(final DeathService deathService, final CcdClientApi ccdC
         this.deathRecordService = deathRecordService;
     }
 
+    public DeathRecord getDeathRecordById(final Integer systemNumber) {
+        log.info("Trying LEV call");
+        V1Death v1Death;
```
Comment (isha2040) [inline src/main/java/uk/gov/hmcts/probate/service/LifeEventService.java]: possibly some more relevant name to the var, we are doing operations on this too

File: src/main/java/uk/gov/hmcts/probate/controller/LifeEventController.java (hunk 1 above)
Comment (nigeldunne) [reply src/main/java/uk/gov/hmcts/probate/controller/LifeEventController.java]: I have now removed this endpoint and implemented it without a second callback

File: src/main/java/uk/gov/hmcts/probate/model/ccd/raw/request/CaseData.java (hunk 2 above)
Comment (nigeldunne) [reply src/main/java/uk/gov/hmcts/probate/model/ccd/raw/request/CaseData.java]: Done

File: src/main/java/uk/gov/hmcts/probate/model/ccd/raw/response/ResponseCaseData.java (hunk 3 above)
Comment (nigeldunne) [reply src/main/java/uk/gov/hmcts/probate/model/ccd/raw/response/ResponseCaseData.java]: These need to be non-final to allow the case data to be updated

File: src/main/java/uk/gov/hmcts/probate/service/DeathRecordService.java (hunk 4 above)
Comment (nigeldunne) [reply src/main/java/uk/gov/hmcts/probate/service/DeathRecordService.java:56]: This warning is becuase of fields in the api annotated as @NotNull. Becuase the api is outside of out control, and may change I think it is safer to keep the null checks and suppress the warning.

File: src/main/java/uk/gov/hmcts/probate/service/LifeEventService.java (hunk 5 above)
Comment (nigeldunne) [reply src/main/java/uk/gov/hmcts/probate/service/LifeEventService.java]: Renamed variable as "record"

**Review by isha2040 — APPROVED** [review]

**Review by isha2040 — APPROVED** [review]

**Comment by nigeldunne** [comment]
Closed - blocked waiting for QA

**Pull request closed without merging**